src/config.rs

```diff
@@ -1494,45 +1494,68 @@ pub enum DBType {
     Raft,
 }
 
-pub struct DBConfigManger {
-    db: RocksEngine,
+pub struct DBConfigManger<T: TabletFactory<RocksEngine>> {
+    tablet_factory: Arc<T>,
     db_type: DBType,
     shared_block_cache: bool,
 }
 
-impl DBConfigManger {
-    pub fn new(db: RocksEngine, db_type: DBType, shared_block_cache: bool) -> Self {
+impl<T: TabletFactory<RocksEngine>> DBConfigManger<T> {
+    pub fn new(tablet_factory: Arc<T>, db_type: DBType, shared_block_cache: bool) -> Self {
         DBConfigManger {
-            db,
+            tablet_factory,
             db_type,
             shared_block_cache,
         }
     }
-}
 
-impl DBConfigManger {
     fn set_db_config(&self, opts: &[(&str, &str)]) -> Result<(), Box<dyn Error>> {
-        self.db.set_db_options(opts)?;
-        Ok(())
+        let mut result = Ok(());
+        self.tablet_factory
+            .loop_tablet_cache(Box::new(|region_id, suffix, db: &RocksEngine| {
+                let r = db.set_db_options(opts);
+                if r.is_err() {
+                    result = Err(Box::from(r.err().unwrap()));
+                    error!(
+                        "set_db_config failed on tablet {}_{}. err:{:?}",
+                        region_id, suffix, &result
+                    );
+                }
+            }));
+        result
     }
 
     fn set_cf_config(&self, cf: &str, opts: &[(&str, &str)]) -> Result<(), Box<dyn Error>> {
+        let mut result = Ok(());
         self.validate_cf(cf)?;
-        self.db.set_options_cf(cf, opts)?;
-        // Write config to metric
-        for (cfg_name, cfg_value) in opts {
-            let cfg_value = match cfg_value {
-                v if *v == "true" => Ok(1f64),
-                v if *v == "false" => Ok(0f64),
-                v => v.parse::<f64>(),
-            };
-            if let Ok(v) = cfg_value {
-                CONFIG_ROCKSDB_GAUGE
-                    .with_label_values(&[cf, cfg_name])
-                    .set(v);
+        self.tablet_factory
+            .loop_tablet_cache(Box::new(|region_id, suffix, db: &RocksEngine| {
+                let r = db.set_options_cf(cf, opts);
+                if r.is_err() {
+                    result = Err(Box::from(r.err().unwrap()));
+                    error!(
+                        "set_cf_config {} failed on tablet {}_{}. err:{:?}",
+                        cf, region_id, suffix, &result
+                    );
+                }
+            }));
+
+        if result.is_ok() {
```

Comment:
i use result?; initially , but it will run into this error.
error[E0283]: type annotations needed
    --> src/config.rs:1535:34
     |
1535 |                     result = Err(Box::from(r.err().unwrap()));
     |                                  ^^^^^^^^^ cannot infer type for struct `Box<_, _>`
     |
     = note: multiple `impl`s satisfying `Box<_, _>: From<engine_traits::Error>` found in the following crates: `alloc`, `std`:
             - impl<'a, E> From<E> for Box<(dyn StdError + 'a)>
               where E: 'a, E: StdError;
             - impl<'a, E> From<E> for Box<(dyn StdError + std::marker::Send + std::marker::Sync + 'a)>
               where E: 'a, E: StdError, E: std::marker::Send, E: std::marker::Sync;
             - impl<T> From<T> for Box<T>;

error[E0283]: type annotations needed
    --> src/config.rs:1544:15
     |
1544 |         result?;
     |               ^ cannot infer type for struct `Box<_, _>`